### PR TITLE
make_generate_sample_default_method

### PR DESCRIPTION
**Pull request by andy-newbranch**
make_generate_sample_default_method

### CODE REVIEW

File: edap/edap.py [hunk 1]
```diff
@@ -151,9 +151,21 @@ def trigger(self, sample: EdapSample) -> EdapSample | None:
...
+        Method for generating base EDAP structure for triggered EDAP samples. The "sensors" and "triggers" properties are empty as these 
+        will be filled in by the various activated triggers. Also "sample_energy" and "duration" will be calcualted if they are not provided.
+        If any calculations involving values from the last sample are needed, this method needs to be overrriden. Any sensors added here will always
+        be present in the triggered EDAP sample.
         """
+        return {
+            "time":  sample.get("time"),
+            "power": sample.get("power"),
+            "energy": sample.get("energy"),
+            "sample_energy": sample.get("sample_energy") if "sample_energy" in sample or sample.get("energy") is None or self._last_sample.get("energy") is None else\
```
Comment (baskalocl) [inline edap/edap.py]: Can we move up this logic? I spent 5mins to figure out, what is logic inside. It will be easier to maintain later.

Comment (andy-newbranch) [reply edap/edap.py]: What do you men "move_up" , You mean to create a local method for it ?

Comment (andy-newbranch) [reply edap/edap.py]: fixed by adding "_has_or_none" local method

Comment (baskalocl) [reply edap/edap.py]: I am not a big fan of an inline if-else clause, so I meant to put the logic up, use some local variable and pass it to resulting dict.

This has_or_none is okay, but still one line if-else is maybe confusing.

Comment (andy-newbranch) [reply edap/edap.py]: ok, maybe something like this then :)

**Review by baskalocl — APPROVED** [review]
Great and helpful!

**Pull request merged**